doc/cephfs/cephfs-mirroring.rst

```diff
@@ -42,129 +44,175 @@ The mirror daemon should be spawned using `systemctl(1)` unit files::
 
-The mirroring module is disabled by default. To enable the mirroring module::
+.. note:: "Mirroring module" commands are prefixed with ``fs snapshot mirror``.
+   This distinguishes them from "monitor commands", which are prefixed with ``fs
+   mirror``. Be sure (in this context) to use module commands.
 
-  $ ceph mgr module enable mirroring
+To disable mirroring for a given file system, run a command of the following form:
 
-The mirroring module provides a family of commands to control mirroring of directory
-snapshots. To add or remove directories, mirroring needs to be enabled for a given
-file system. To enable mirroring for a given file system::
+.. prompt:: bash $
 
-  $ ceph fs snapshot mirror enable <fs_name>
+   ceph fs snapshot mirror disable <fs_name>
 
-.. note:: Mirroring module commands are prefixed with `fs snapshot mirror` as compared to
-          monitor commands which are prefixed with `fs mirror`. Be sure to use module
-          commands.
+After mirroring is enabled, add a peer to which directory snapshots are to be
+mirrored. Peers are specified by the ``<client>@<cluster>`` format, which is
+referred to elsewhere in this document as the ``remote_cluster_spec``. Peers
+are assigned a unique-id (UUID) when added. See the :ref:`Creating
+Users<cephfs_mirroring_creating_users>` section for instructions that describe
+how to create Ceph users for mirroring.
 
-To disable mirroring for a given file system::
+To add a peer, run a command of the following form:
 
-  $ ceph fs snapshot mirror disable <fs_name>
+.. prompt:: bash $
 
-Once mirroring is enabled, add a peer to which directory snapshots are to be mirrored.
-Peers are specified by `<client>@<cluster>` format which is also termed as `remote_cluster_spec`
-(further in this document) and are assigned a unique-id (UUID) when added. See `Creating Users`
-section on how to create Ceph users for mirroring.
+   ceph fs snapshot mirror peer_add <fs_name> <remote_cluster_spec> [<remote_fs_name>] [<remote_mon_host>] [<cephx_key>]
 
-To add a peer use::
+``<remote_cluster_spec>`` is of the format ``client.<id>@<cluster_name>``.
 
-  $ ceph fs snapshot mirror peer_add <fs_name> <remote_cluster_spec> [<remote_fs_name>] [<remote_mon_host>] [<cephx_key>]
+``<remote_fs_name>`` is optional, and defaults to `<fs_name>` (on the remote
+cluster).
 
-`<remote_cluster_spec>` is of format `client.<id>@<cluster_name>`.
-`<remote_fs_name>` is optional, and defaults to `<fs_name>` (on the remote cluster).
+For this command to succeed, the remote cluster's Ceph configuration and user
+keyring must be available in the primary cluster. For example, if a user named
+``client_mirror`` is created on the remote cluster which has ``rwps``
+permissions for the remote file system named ``remote_fs`` (see `Creating
+Users`) and the remote cluster is named ``remote_ceph`` (that is, the remote
+cluster configuration file is named ``remote_ceph.conf`` on the primary
```

Comment:
Ambiguity here caused people in the past to think they needed custom cluster names with rbd-mirror.  Nice to see this made explicit.